### PR TITLE
Port test_ops.py to pytest

### PR DESCRIPTION
**Pull request by NicolasHug**
This PR ports the entire `test_ops.py`file to pytest. It went pretty smoothly considering the size of the file, as the class structure already in place really helped.

The diff looks a bit big but it's mainly because I parametrized some methods, so the indentation of the (now removed) for loops isn't here anymore.

The rest is just adding a bunch of decorators and changing class names so that they're properly discovered by pytest

### CODE REVIEW

File: test/test_ops.py [hunk 1]
```diff
@@ -15,48 +14,12 @@
...
-
-    def _test_backward(self, device, contiguous):
-        pass
-
-
-class RoIOpTester(OpTester):
-    def _test_forward(self, device, contiguous, x_dtype=None, rois_dtype=None, **kwargs):
+    @pytest.mark.parametrize('device', cpu_and_gpu())
+    @pytest.mark.parametrize('contiguous', (True, False))
+    def test_forward(self, device, contiguous, x_dtype=None, rois_dtype=None, **kwargs):
```
Comment (NicolasHug) [inline test/test_ops.py:22]: The previously manual cross-product between device and contiguous is now seamlessly handled by the parametrization, both for `test_forward` and `test_backward`

File: test/test_ops.py [hunk 2]
```diff
@@ -436,7 +416,7 @@ def test_msroialign_repr(self):
         # Check integrity of object __repr__ attribute
         expected_string = (f"MultiScaleRoIAlign(featmap_names={fmap_names}, output_size={output_size}, "
                            f"sampling_ratio={sampling_ratio})")
-        self.assertEqual(t.__repr__(), expected_string)
+        assert t.__repr__() == expected_string
```
Comment (pmeier) [inline test/test_ops.py:419]: Why not ?

```suggestion
        assert repr(t) == expected_string
```

Can be applied multiple times.

File: test/test_ops.py [hunk 3]
```diff
@@ -943,7 +936,8 @@ def test_bbox_convert_jit(self):
         torch.testing.assert_close(scripted_cxcywh, box_cxcywh, rtol=0.0, atol=TOLERANCE)
 
 
-class BoxAreaTester(unittest.TestCase):
+@cpu_only
+class BoxAreaTester:
```
Comment (pmeier) [inline test/test_ops.py:940]: Shouldn't this be 
```suggestion
class TestBoxArea:
```
?

**Review by pmeier — COMMENTED** [review]
Looking good @NicolasHug. Two comments:

- Can we declare `RoIOpTester` an `abc.ABC` and mark the abstract methods as such? Makes it easier to read and catches errors faster if we need to add a new test case.
- Can the `_test_boxes_shape` merged into the base class with the same pattern as the other tests?

**Comment by NicolasHug** [comment]
Thanks for the review @pmeier ! Your comments were addressed :)

> _test_boxes_shape merged into the base class with the same pattern as the other tests?

Yeah it was a bit weird. I just kept `    def _helper_boxes_shape(self, func):` in the base class and declared test functions in the child classes as 

```py
    @cpu_only
    def test_boxes_shape(self):
        self._helper_boxes_shape(ops.roi_pool)
```

**Review by pmeier — APPROVED** [review]
LGTM, thanks @NicolasHug!

**Pull request merged**